src/ConfigurationManager.cpp

```
@@ -61,6 +63,64 @@ void sanitizeHostname(char* buf, size_t cap) {
     buf[cap - 1] = '\0';
 }
 
+// Reject GPIOs that can't safely drive the status LED on the CURRENT board and
+// fall back to the default sentinel (LED_PIN_DEFAULT -> PIN_STATUS_LED). Blocklists
+// are selected by the same BOARD_* flags as BoardPins.h. Used as an NVS boundary
+// guard on both read (loadFromNVS) and write (saveToNVS), same as sanitizeHostname.
+static uint8_t sanitizeLedPin(uint8_t pin) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)